### ISSUE TITLE
[java] ImmutableField false-positive with lambdas

### ISSUE THREAD

**Issue by boris-petrov** [pmd/pmd#855]
PMD version `6.0.0`. Something like:

```java
class Bar {
}

class Foo {
	private Bar bar;

	public Foo() {
		someMethod(() -> {
			bar = new Bar();
		});
	}

	private void someMethod(Runnable action) {
		action.run();
	}
}
```

Gives an `ImmutableField` warning for `bar` which actually is not correct - putting `final` on `bar` prevents that code from compiling.
